### DISCUSSION TITLE
Helm Chart for Self Hosted Deployment in Kubernetes

### DISCUSSION THREAD

**Discussion by jugrajsingh** [Shelf-nu/shelf.nu#887] [Ideas]
I was wondering if anyone is working on a deployment template that would provide an easy way to deploy this in a Kubernetes Environment.

Following will be the Features of that:
- [ ] Ability to Install into a Kubernetes Cluster from the Public Docker in One Click
- [ ] Ability to Chose External Supabase or Self-Host one in the same command.
- [ ] Ability to Check if the Database is initialized and trigger the setup and migration on a new release.

If we can figure out the Database Initialization on every start, I can help create a draft Helm Chart to deploy the docker.

Currently, before a new version is released, I have to run a setup command outside the docker to ensure DB is migrated.

**Comment by DonKoko** [comment]
Hey @jugrajsingh . Thanks for your suggestion and opening a discussion.
This is not something we have prioritized or even though about before, however we are willing to assist in doing something like this.

Lets wait and see of there are more people interested in it. 
As for this:
> Currently, before a new version is released, I have to run a setup command outside the docker to ensure DB is migrated.

I am not sure whats the best way to resolve this. We did release some update to our docker setup that allows for running the migrations. I am not sure if it can be automated or it has to be done manually, as docker is community maintained and we don't use it in our team.
